libtiledbsoma/src/utils/arrow_adapter.cc

```diff
@@ -1169,84 +1203,28 @@ ArraySchema ArrowAdapter::tiledb_schema_from_arrow_schema(
     CurrentDomain current_domain(*ctx);
     NDRectangle ndrect(*ctx, domain);
 
-    for (int64_t sch_idx = 0; sch_idx < arrow_schema->n_children; ++sch_idx) {
-        auto child = arrow_schema->children[sch_idx];
-        auto type = ArrowAdapter::to_tiledb_format(child->format);
+    for (auto column : columns) {
+        if (!column->isIndexColumn()) {
+            continue;
+        }
 
-        for (int64_t i = 0; i < index_column_schema->n_children; ++i) {
-            auto col_name = index_column_schema->children[i]->name;
-            if (strcmp(child->name, col_name) != 0) {
-                continue;
+        if (column->name() == SOMA_GEOMETRY_COLUMN_NAME) {
+            std::vector<std::any> dom;
```

Comment:
As on previous PRs: we cannot simply say `dom` or `domain`, ever.

There are four things it can mean:

* core domain (which is soma maxdomain)
* soma domain (which is core current domain)

The names are confusing (and too late to change), and confusion is too easy, and developer confusion is high-risk

Reply:
Perhaps rename `dom` to `cdslot`